src/base_directories.rs

```diff
@@ -952,6 +992,253 @@ mod test {
         });
     }
 
```

Comment:
Added these public tests to cover the Public API to prevent regression and breaking changes.